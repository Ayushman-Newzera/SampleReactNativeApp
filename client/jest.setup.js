import {NativeModules} from 'react-native';
// Mock the ImagePickerManager native module to allow us to unit test the JavaScript code
NativeModules.ImagePickerManager = {
  // eslint-disable-next-line no-undef
  showImagePicker: jest.fn(),
  // eslint-disable-next-line no-undef
  launchCamera: jest.fn(),
  // eslint-disable-next-line no-undef
  launchImageLibrary: jest.fn(),
};

NativeModules.RNGestureHandlerModule = {
  // eslint-disable-next-line no-undef
  attachGestureHandler: jest.fn(),
  // eslint-disable-next-line no-undef
  createGestureHandler: jest.fn(),
  // eslint-disable-next-line no-undef
  dropGestureHandler: jest.fn(),
  // eslint-disable-next-line no-undef
  updateGestureHandler: jest.fn(),
  State: {},
  Directions: {},
};

// Reset the mocks before each test
global.beforeEach(() => {
  // eslint-disable-next-line no-undef
  jest.clearAllMocks();
});
