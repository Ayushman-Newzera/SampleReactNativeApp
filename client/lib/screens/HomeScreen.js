var _interopRequireDefault = require('@babel/runtime/helpers/interopRequireDefault');
Object.defineProperty(exports, '__esModule', {value: true});
exports.default = void 0;
var _regenerator = _interopRequireDefault(
  require('@babel/runtime/regenerator'),
);
var _toConsumableArray2 = _interopRequireDefault(
  require('@babel/runtime/helpers/toConsumableArray'),
);
var _slicedToArray2 = _interopRequireDefault(
  require('@babel/runtime/helpers/slicedToArray'),
);
var _taggedTemplateLiteralLoose2 = _interopRequireDefault(
  require('@babel/runtime/helpers/taggedTemplateLiteralLoose'),
);
var _client = require('@apollo/client');
var _asyncStorage = _interopRequireDefault(
  require('@react-native-async-storage/async-storage'),
);
var _native = require('@react-navigation/native');
var _react = _interopRequireWildcard(require('react'));
var _reactNative = require('react-native');
var _templateObject,
  _templateObject2,
  _jsxFileName =
    '/Users/ayushman/Desktop/SampleReactNativeApp1/client/src/screens/HomeScreen.js';
function _getRequireWildcardCache(nodeInterop) {
  if (typeof WeakMap !== 'function') {
    return null;
  }
  var cacheBabelInterop = new WeakMap();
  var cacheNodeInterop = new WeakMap();
  return (_getRequireWildcardCache = function _getRequireWildcardCache(
    nodeInterop,
  ) {
    return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
  })(nodeInterop);
}
function _interopRequireWildcard(obj, nodeInterop) {
  if (!nodeInterop && obj && obj.__esModule) {
    return obj;
  }
  if (obj === null || (typeof obj !== 'object' && typeof obj !== 'function')) {
    return {default: obj};
  }
  var cache = _getRequireWildcardCache(nodeInterop);
  if (cache && cache.has(obj)) {
    return cache.get(obj);
  }
  var newObj = {};
  var hasPropertyDescriptor =
    Object.defineProperty && Object.getOwnPropertyDescriptor;
  for (var key in obj) {
    if (key !== 'default' && Object.prototype.hasOwnProperty.call(obj, key)) {
      var desc = hasPropertyDescriptor
        ? Object.getOwnPropertyDescriptor(obj, key)
        : null;
      if (desc && (desc.get || desc.set)) {
        Object.defineProperty(newObj, key, desc);
      } else {
        newObj[key] = obj[key];
      }
    }
  }
  newObj.default = obj;
  if (cache) {
    cache.set(obj, newObj);
  }
  return newObj;
}
var USER_DETAILS = (0, _client.gql)(
  _templateObject ||
    (_templateObject = (0, _taggedTemplateLiteralLoose2.default)([
      '\n  query UserDetails {\n    getUserDetails {\n      name\n      bio\n      profileImageLink\n    }\n  }\n',
    ])),
);
var UPDATE_PROFILE = (0, _client.gql)(
  _templateObject2 ||
    (_templateObject2 = (0, _taggedTemplateLiteralLoose2.default)([
      '\n  mutation UpdateProfileMutation($profileImageLink: String!) {\n    addProfilePicture(profileImageLink: $profileImageLink)\n  }\n',
    ])),
);
function HomeScreen(_ref) {
  var navigation = _ref.navigation;
  var _useState = (0, _react.useState)(true),
    _useState2 = (0, _slicedToArray2.default)(_useState, 2),
    borderActive = _useState2[0],
    setBorderActive = _useState2[1];
  var anonymousImage = {
    uri: 'https://image.shutterstock.com/image-illustration/photo-silhouette-male-profile-white-260nw-1018631086.jpg',
  };
  var _useState3 = (0, _react.useState)(anonymousImage),
    _useState4 = (0, _slicedToArray2.default)(_useState3, 2),
    profilePicture = _useState4[0],
    setProfilePicture = _useState4[1];
  var _useState5 = (0, _react.useState)([
      {
        imageLink:
          'https://images.unsplash.com/photo-1584714268709-c3dd9c92b378?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1099&q=80',
      },
    ]),
    _useState6 = (0, _slicedToArray2.default)(_useState5, 2),
    stories = _useState6[0],
    setStories = _useState6[1];
  var _useQuery = (0, _client.useQuery)(USER_DETAILS),
    loading = _useQuery.loading,
    error = _useQuery.error,
    data = _useQuery.data;
  var _useMutation = (0, _client.useMutation)(UPDATE_PROFILE),
    _useMutation2 = (0, _slicedToArray2.default)(_useMutation, 1),
    addProfileImageLink = _useMutation2[0];
  var isFocused = (0, _native.useIsFocused)();
  var handleBorderActivity = function handleBorderActivity() {
    setBorderActive(false);
  };
  (0, _react.useEffect)(
    function () {
      getDataProfile();
      getDataStory();
    },
    [isFocused],
  );
  var buttonClickedHandler = function buttonClickedHandler() {
    navigation.navigate('Stories Input');
  };
  var getDataProfile = function getDataProfile() {
    var jsonValue;
    return _regenerator.default.async(
      function getDataProfile$(_context) {
        while (1) {
          switch ((_context.prev = _context.next)) {
            case 0:
              _context.prev = 0;
              _context.next = 3;
              return _regenerator.default.awrap(
                _asyncStorage.default.getItem('@profilePicture'),
              );
            case 3:
              jsonValue = _context.sent;
              if (!(jsonValue != null && jsonValue !== undefined)) {
                _context.next = 9;
                break;
              }
              _context.next = 7;
              return _regenerator.default.awrap(
                addProfileImageLink({
                  variables: {
                    profileImageLink: JSON.stringify(JSON.parse(jsonValue)),
                  },
                })
                  .then(function (res) {
                    setProfilePicture(JSON.parse(res.data.addProfilePicture));
                  })
                  .catch(function (error) {
                    console.log('???', error);
                  }),
              );
            case 7:
              _context.next = 10;
              break;
            case 9:
              setProfilePicture(anonymousImage);
            case 10:
              _context.next = 15;
              break;
            case 12:
              _context.prev = 12;
              _context.t0 = _context.catch(0);
              console.log('error while getDataProfile', _context.t0);
            case 15:
            case 'end':
              return _context.stop();
          }
        }
      },
      null,
      null,
      [[0, 12]],
      Promise,
    );
  };
  var getDataStory = function getDataStory() {
    var jsonValue, value, val;
    return _regenerator.default.async(
      function getDataStory$(_context2) {
        while (1) {
          switch ((_context2.prev = _context2.next)) {
            case 0:
              _context2.prev = 0;
              _context2.next = 3;
              return _regenerator.default.awrap(
                _asyncStorage.default.getItem('@storyPicture'),
              );
            case 3:
              jsonValue = _context2.sent;
              if (jsonValue != null && jsonValue !== undefined) {
                value = JSON.parse(jsonValue).uri;
                val = 0;
                stories.map(function (item) {
                  if (item.imageLink === value) {
                    val += 1;
                  }
                });
                if (val == 0) {
                  setStories(function (prev) {
                    return [].concat((0, _toConsumableArray2.default)(prev), [
                      {imageLink: value},
                    ]);
                  });
                  setBorderActive(true);
                }
              }
              _context2.next = 10;
              break;
            case 7:
              _context2.prev = 7;
              _context2.t0 = _context2.catch(0);
              console.log('error while getDataStory', _context2.t0);
            case 10:
            case 'end':
              return _context2.stop();
          }
        }
      },
      null,
      null,
      [[0, 7]],
      Promise,
    );
  };
  if (loading) {
    return _react.default.createElement(
      _reactNative.Text,
      {
        __self: this,
        __source: {fileName: _jsxFileName, lineNumber: 125, columnNumber: 12},
      },
      "'Loading...'",
    );
  }
  if (error) {
    return _react.default.createElement(
      _reactNative.Text,
      {
        __self: this,
        __source: {fileName: _jsxFileName, lineNumber: 128, columnNumber: 12},
      },
      '`Error! WTF is $',
      error.message,
      '`',
    );
  }
  return _react.default.createElement(
    _reactNative.View,
    {
      style: {flex: 1, alignItems: 'center', justifyContent: 'center'},
      __self: this,
      __source: {fileName: _jsxFileName, lineNumber: 134, columnNumber: 5},
    },
    _react.default.createElement(
      _reactNative.View,
      {
        style: styles.profileContainer,
        __self: this,
        __source: {fileName: _jsxFileName, lineNumber: 135, columnNumber: 7},
      },
      _react.default.createElement(
        _reactNative.TouchableOpacity,
        {
          onPress: function onPress() {
            navigation.navigate('Stories', {
              stories: {stories: stories},
              handleBorderActivity: {
                handleBorderActivity: handleBorderActivity,
              },
            });
          },
          onLongPress: function onLongPress() {
            navigation.navigate('Profile Input');
          },
          style: styles.profile,
          __self: this,
          __source: {fileName: _jsxFileName, lineNumber: 136, columnNumber: 9},
        },
        _react.default.createElement(_reactNative.Image, {
          style: [
            styles.container,
            {borderColor: borderActive ? '#FAD02C' : 'grey'},
          ],
          source: profilePicture,
          __self: this,
          __source: {fileName: _jsxFileName, lineNumber: 147, columnNumber: 11},
        }),
      ),
      _react.default.createElement(
        _reactNative.View,
        {
          style: styles.roundButton,
          __self: this,
          __source: {fileName: _jsxFileName, lineNumber: 157, columnNumber: 9},
        },
        _react.default.createElement(
          _reactNative.TouchableOpacity,
          {
            onPress: buttonClickedHandler,
            style: styles.roundButton1,
            __self: this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 158,
              columnNumber: 11,
            },
          },
          _react.default.createElement(
            _reactNative.View,
            {
              __self: this,
              __source: {
                fileName: _jsxFileName,
                lineNumber: 161,
                columnNumber: 13,
              },
            },
            _react.default.createElement(
              _reactNative.Text,
              {
                __self: this,
                __source: {
                  fileName: _jsxFileName,
                  lineNumber: 162,
                  columnNumber: 15,
                },
              },
              '+',
            ),
          ),
        ),
      ),
    ),
    _react.default.createElement(
      _reactNative.View,
      {
        __self: this,
        __source: {fileName: _jsxFileName, lineNumber: 168, columnNumber: 7},
      },
      _react.default.createElement(
        _reactNative.Text,
        {
          __self: this,
          __source: {fileName: _jsxFileName, lineNumber: 169, columnNumber: 9},
        },
        data.getUserDetails.name,
      ),
    ),
    _react.default.createElement(
      _reactNative.View,
      {
        style: styles.bio,
        __self: this,
        __source: {fileName: _jsxFileName, lineNumber: 171, columnNumber: 7},
      },
      _react.default.createElement(
        _reactNative.Text,
        {
          __self: this,
          __source: {fileName: _jsxFileName, lineNumber: 172, columnNumber: 9},
        },
        data.getUserDetails.bio,
      ),
    ),
  );
}
var styles = _reactNative.StyleSheet.create({
  container: {
    height: 200,
    width: 200,
    borderRadius: 100,
    borderColor: '#FAD02C',
    borderWidth: 5,
    marginBottom: 50,
    alignItems: 'center',
  },
  image: {height: 200, width: 200},
  roundButton: {justifyContent: 'center', alignItems: 'center'},
  roundButton1: {
    width: 30,
    height: 30,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 100,
    backgroundColor: 'orange',
    position: 'relative',
    marginTop: 70,
    marginRight: 100,
  },
  profileContainer: {flexDirection: 'row', position: 'relative'},
  profile: {flex: 1, position: 'relative', marginLeft: 96},
  bio: {justifyContent: 'center', alignItems: 'center'},
});
var _default = HomeScreen;
exports.default = _default;
