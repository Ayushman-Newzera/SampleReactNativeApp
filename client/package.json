{
  "name": "client",
  "version": "0.0.1",
  "private": true,
  "main": "lib/index.js",
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "start": "react-native start",
    "test": "jest",
    "lint": "eslint .",
    "build": "babel src/ -d lib/",
    "prepublish": "npm run build",
    "flow": "flow"
  },
  "dependencies": {
    "@apollo/client": "^3.3.20",
    "@babel/preset-env": "^7.14.7",
    "@babel/preset-react": "^7.14.5",
    "@react-native-async-storage/async-storage": "github:react-native-async-storage/async-storage",
    "@react-native-community/masked-view": "^0.1.11",
    "@react-navigation/native": "^5.9.4",
    "@react-navigation/stack": "^5.14.5",
    "graphql": "^15.5.1",
    "prop-types": "^15.7.2",
    "react": "16.9.0",
    "react-dom": "^16.9.0",
    "react-native": "0.64.2",
    "react-native-gesture-handler": "^1.10.3",
    "react-native-image-picker": "^2.3.4",
    "react-native-reanimated": "^2.2.0",
    "react-native-safe-area-context": "^3.2.0",
    "react-native-screens": "^3.4.0"
  },
  "devDependencies": {
    "@babel/cli": "^7.14.5",
    "@babel/core": "^7.14.6",
    "@babel/preset-flow": "^7.14.5",
    "@babel/runtime": "^7.14.6",
    "@react-native-community/eslint-config": "^3.0.0",
    "@wojtekmaj/enzyme-adapter-react-17": "^0.6.2",
    "babel-jest": "^27.0.6",
    "enzyme": "^3.11.0",
    "enzyme-adapter-react-16": "^1.15.6",
    "enzyme-to-json": "^3.6.2",
    "eslint": "^7.29.0",
    "flow-bin": "^0.154.0",
    "gulp-webserver": "^0.9.1",
    "jest": "^26.6.0",
    "metro-react-native-babel-preset": "^0.66.0",
    "react-native-testing-library": "^6.0.0",
    "react-test-renderer": "^17.0.2"
  },
  "jest": {
    "preset": "react-native",
    "verbose": true,
    "snapshotSerializers": [
      "enzyme-to-json/serializer"
    ],
    "setupFiles": [
      "./__tests__/Setup.js"
    ],
    "setupFilesAfterEnv": [
      "<rootDir>/jest.setup.js"
    ],
    "testPathIgnorePatterns": [
      "/node_modules/",
      "<rootDir>/__tests__/Setup.js"
    ],
    "testEnvironment": "jsdom"
  }
}
