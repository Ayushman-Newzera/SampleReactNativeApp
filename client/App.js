/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 *
 */

import {
  ApolloClient,
  ApolloProvider,
  createHttpLink,
  InMemoryCache,
} from '@apollo/client';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import React from 'react';
import HomeScreen from './src/screens/HomeScreen';
import ProfilePictureInput from './src/screens/ProfilePictureInput';
import StoryInputScreen from './src/screens/StoryInputScreen';
import StoryScreen from './src/screens/StoryScreen';

const httpLink = createHttpLink({
  uri: 'http://192.168.1.35:4000',
});

export const client = new ApolloClient({
  link: httpLink,
  cache: new InMemoryCache(),
});

const Stack = createStackNavigator();

function App() {
  return (
    <ApolloProvider client={client}>
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen name="Home" component={HomeScreen} />
          <Stack.Screen name="Stories" component={StoryScreen} />
          <Stack.Screen name="Stories Input" component={StoryInputScreen} />
          <Stack.Screen name="Profile Input" component={ProfilePictureInput} />
        </Stack.Navigator>
      </NavigationContainer>
    </ApolloProvider>
  );
}

export default App;
