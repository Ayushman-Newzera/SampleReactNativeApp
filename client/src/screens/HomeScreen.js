// @flow

import {gql, useMutation, useQuery} from '@apollo/client';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {useIsFocused} from '@react-navigation/native';
import React, {useEffect, useState} from 'react';
import {Image, StyleSheet, Text, TouchableOpacity, View} from 'react-native';
// import {client} from '../../App';

export const USER_DETAILS = gql`
  query UserDetails {
    getUserDetails {
      name
      bio
      profileImageLink
    }
  }
`;

export const UPDATE_PROFILE = gql`
  mutation UpdateProfileMutation($profileImageLink: String) {
    addProfilePicture(profileImageLink: $profileImageLink)
  }
`;

// type HomeScreenNavigationProp = StackNavigationProp<{Home: undefined}, 'Home'>;

// type HomeScreenProps = {
//   navigation: HomeScreenNavigationProp,
// };

// type HomeScreenProps = {
//   navigation: PropTypes.object,
// };

function HomeScreen({navigation}) {
  // const {navigation} = props;
  const [borderActive, setBorderActive] = useState(true);

  const anonymousImage = {
    uri: 'https://image.shutterstock.com/image-illustration/photo-silhouette-male-profile-white-260nw-1018631086.jpg',
  };

  const [profilePicture, setProfilePicture] = useState(anonymousImage);

  const [stories, setStories] = useState([
    {
      imageLink:
        'https://images.unsplash.com/photo-1584714268709-c3dd9c92b378?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1099&q=80',
    },
  ]);

  const {loading, error, data} = useQuery(USER_DETAILS);

  const [addProfileImageLink] = useMutation(UPDATE_PROFILE); // Hooks should not be rendered conditionally

  const isFocused = useIsFocused();

  const handleBorderActivity = () => {
    setBorderActive(false);
  };

  useEffect(() => {
    getDataProfile();
    getDataStory();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [isFocused]);

  const buttonClickedHandler = () => {
    navigation.navigate('Stories Input');
  };

  const getDataProfile = async () => {
    try {
      const jsonValue = await AsyncStorage.getItem('@profilePicture');

      if (jsonValue != null && jsonValue !== undefined) {
        // console.log('string', JSON.stringify(JSON.parse(jsonValue)));
        await addProfileImageLink({
          variables: {
            profileImageLink: JSON.stringify(JSON.parse(jsonValue)),
          },
        })
          .then(res => {
            setProfilePicture(JSON.parse(res.data.addProfilePicture));
          })
          // eslint-disable-next-line no-shadow
          .catch(error => {
            console.log(error);
          });
      } else {
        setProfilePicture(anonymousImage);
      }
      // return jsonValue != null ? JSON.parse(jsonValue) : null;
    } catch (e) {
      // error reading value
    }
  };

  const getDataStory = async () => {
    try {
      const jsonValue = await AsyncStorage.getItem('@storyPicture');

      if (jsonValue != null && jsonValue !== undefined) {
        const value = JSON.parse(jsonValue).uri;
        let val = 0;
        stories.map(item => {
          if (item.imageLink === value) {
            val += 1;
          }
        });
        // eslint-disable-next-line eqeqeq
        if (val == 0) {
          setStories(prev => [...prev, {imageLink: value}]);
          setBorderActive(true);
        }
      }
      // return jsonValue != null ? jsonValue : null;
    } catch (e) {
      // error reading value
    }
  };

  // useEffect(() => {
  //   console.log(loading, error, data);
  // }, [loading, error, data]);

  if (loading) {
    return <Text>Loading...</Text>;
  }

  if (error) {
    return <Text>`Error! WTF is ${error.message}`</Text>;
  }

  return (
    // <ApolloProvider client={client}>
    // eslint-disable-next-line react-native/no-inline-styles
    <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
      <View style={styles.profileContainer}>
        <TouchableOpacity
          onPress={() => {
            navigation.navigate('Stories', {
              stories: {stories},
              handleBorderActivity: {handleBorderActivity},
            });
          }}
          onLongPress={() => {
            navigation.navigate('Profile Input');
          }}
          style={styles.profile}>
          <Image
            style={[
              styles.container,
              // eslint-disable-next-line react-native/no-inline-styles
              {borderColor: borderActive ? '#FAD02C' : 'grey'},
            ]}
            source={profilePicture}
          />
        </TouchableOpacity>

        <View style={styles.roundButton}>
          <TouchableOpacity
            onPress={buttonClickedHandler}
            style={styles.roundButton1}>
            <View>
              <Text>+</Text>
            </View>
          </TouchableOpacity>
        </View>
      </View>

      <View>
        <Text>{data.getUserDetails.name}</Text>
      </View>
      <View style={styles.bio}>
        <Text>{data.getUserDetails.bio}</Text>
      </View>
    </View>
    // </ApolloProvider>
  );
}

const styles = StyleSheet.create({
  container: {
    height: 200,
    width: 200,
    borderRadius: 100,
    borderColor: '#FAD02C',
    borderWidth: 5,
    marginBottom: 50,
    alignItems: 'center',
  },
  image: {
    height: 200,
    width: 200,
  },
  roundButton: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  roundButton1: {
    width: 30,
    height: 30,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 100,
    backgroundColor: 'orange',
    position: 'relative',
    marginTop: 70,
    marginRight: 100,
  },
  profileContainer: {
    flexDirection: 'row',
    position: 'relative',
  },
  profile: {
    flex: 1,
    position: 'relative',
    marginLeft: 96,
  },
  bio: {
    justifyContent: 'center',
    alignItems: 'center',
  },
});

// HomeScreen.PropTypes = {
//   navigation: PropTypes.array,
// };

export default HomeScreen;
