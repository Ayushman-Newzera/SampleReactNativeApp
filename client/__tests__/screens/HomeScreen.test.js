import {MockedProvider} from '@apollo/client/testing';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {useIsFocused} from '@react-navigation/native';
import {mount} from 'enzyme';
import React from 'react';
import HomeScreen, {
  UPDATE_PROFILE,
  USER_DETAILS,
} from '../../src/screens/HomeScreen';
const wait = require('waait');

const createTestProps = props => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...props,
});

const mocks = [
  {
    request: {
      query: USER_DETAILS,
    },
    result: {
      data: {
        getUserDetails: {
          id: 1,
          name: 'Byoung ho',
          bio: 'Photographer',
          profileImageLink:
            '{"uri":"file:///storage/emulated/0/Pictures/image-60611026-7ced-499c-972a-fb945c72120f.jpg"}',
        },
      },
    },
  },
  {
    request: {
      query: UPDATE_PROFILE,
      variables: {
        profileImageLink:
          '{"uri":"file:///storage/emulated/0/Pictures/image-820516d8-2143-4614-8c60-fb9862e6587a.jpg"}',
      },
    },
    result: () => {
      //   console.log('mutation called');
      return {
        data: {
          addProfilePicture:
            '{"uri":"file:///storage/emulated/0/Pictures/image-820516d8-2143-4614-8c60-fb9862e6587a.jpg"}',
        },
      };
    },
  },
];

describe('HomeScreen', () => {
  it('should render loading state initially', () => {
    const props = createTestProps({});
    const wrapper = mount(
      <MockedProvider mocks={[]} addTypename={false}>
        <HomeScreen {...props} />
      </MockedProvider>,
    );

    expect(wrapper.text()).toBe('Loading...');
  });

  it('should render data after loading', async () => {
    const props = createTestProps({});
    const wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <HomeScreen {...props} />
      </MockedProvider>,
    );

    await wait(0);
    wrapper.update();

    expect(wrapper.get(0)).toMatchSnapshot();

    expect(wrapper.text()[0]).toBe('+');
  });

  it('should show error UI', async () => {
    const props = createTestProps({});

    const userDetailsMock = [
      {
        request: {
          query: USER_DETAILS,
        },
        error: new Error('here goes the error'),
      },
    ];

    const wrapper = mount(
      <MockedProvider mocks={userDetailsMock} addTypename={false}>
        <HomeScreen {...props} />
      </MockedProvider>,
    );

    await wait(0);
    wrapper.update();

    expect(wrapper.text()).toBe('`Error! WTF is $here goes the error`');
  });

  it('should handle navigation correctly', async () => {
    const props = createTestProps({});
    const wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <HomeScreen {...props} />
      </MockedProvider>,
    );

    await wait(0);
    wrapper.update();

    const touchableOpacity1 = wrapper.find('TouchableOpacity').get(0);
    touchableOpacity1.props.onPress();
    wrapper.update();
    expect(props.navigation.navigate).toHaveBeenCalledTimes(1);

    touchableOpacity1.props.onLongPress();
    wrapper.update();
    expect(props.navigation.navigate).toHaveBeenCalledWith('Profile Input');

    // console.log(wrapper.find('TouchableOpacity').get(1));
    const touchableOpacity2 = wrapper.find('TouchableOpacity').get(1);
    touchableOpacity2.props.onPress();
    wrapper.update();
    expect(props.navigation.navigate).toHaveBeenCalledWith('Stories Input');
  });

  it('getDataStory and getDataStory should work correctly', async () => {
    AsyncStorage.getItem = jest.fn();

    useIsFocused.mockImplementation(() => true);

    AsyncStorage.getItem.mockImplementation(arg => {
      if (arg === '@profilePicture') {
        return '{"uri":"file:///storage/emulated/0/Pictures/image-820516d8-2143-4614-8c60-fb9862e6587a.jpg"}';
      } else if (arg === '@storyPicture') {
        return '{"uri":"storyPicture"}';
      }
    });

    JSON.stringify = jest.fn();
    JSON.stringify.mockImplementation(
      () =>
        '{"uri":"file:///storage/emulated/0/Pictures/image-820516d8-2143-4614-8c60-fb9862e6587a.jpg"}',
    );

    const props = createTestProps({});
    const wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <HomeScreen {...props} />
      </MockedProvider>,
    );

    await wait(100);
    wrapper.update();
    // console.log(wrapper.get(0));
    expect(AsyncStorage.getItem).toHaveBeenCalledWith('@profilePicture');
    expect(AsyncStorage.getItem).toHaveBeenCalledWith('@storyPicture');
    // const jsonValue = AsyncStorage.getItem('@profilePicture');
    // console.log(jsonValue);
  });
});
