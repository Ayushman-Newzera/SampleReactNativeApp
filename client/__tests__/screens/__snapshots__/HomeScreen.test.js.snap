// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen should render data after loading 1`] = `
<MockedProvider
  addTypename={false}
  mocks={
    Array [
      Object {
        "request": Object {
          "query": Object {
            "definitions": Array [
              Object {
                "directives": Array [],
                "kind": "OperationDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "UserDetails",
                },
                "operation": "query",
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "getUserDetails",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "name",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "bio",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "profileImageLink",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
                "variableDefinitions": Array [],
              },
            ],
            "kind": "Document",
            "loc": Object {
              "end": 98,
              "start": 0,
            },
          },
        },
        "result": Object {
          "data": Object {
            "getUserDetails": Object {
              "bio": "Photographer",
              "id": 1,
              "name": "Byoung ho",
              "profileImageLink": "{\\"uri\\":\\"file:///storage/emulated/0/Pictures/image-60611026-7ced-499c-972a-fb945c72120f.jpg\\"}",
            },
          },
        },
      },
      Object {
        "request": Object {
          "query": Object {
            "definitions": Array [
              Object {
                "directives": Array [],
                "kind": "OperationDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "UpdateProfileMutation",
                },
                "operation": "mutation",
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [
                        Object {
                          "kind": "Argument",
                          "name": Object {
                            "kind": "Name",
                            "value": "profileImageLink",
                          },
                          "value": Object {
                            "kind": "Variable",
                            "name": Object {
                              "kind": "Name",
                              "value": "profileImageLink",
                            },
                          },
                        },
                      ],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "addProfilePicture",
                      },
                      "selectionSet": undefined,
                    },
                  ],
                },
                "variableDefinitions": Array [
                  Object {
                    "defaultValue": undefined,
                    "directives": Array [],
                    "kind": "VariableDefinition",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "String",
                      },
                    },
                    "variable": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "profileImageLink",
                      },
                    },
                  },
                ],
              },
            ],
            "kind": "Document",
            "loc": Object {
              "end": 126,
              "start": 0,
            },
          },
          "variables": Object {
            "profileImageLink": "{\\"uri\\":\\"file:///storage/emulated/0/Pictures/image-820516d8-2143-4614-8c60-fb9862e6587a.jpg\\"}",
          },
        },
        "result": [Function],
      },
    ]
  }
>
  <HomeScreen
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
  />
</MockedProvider>
`;
